packages/grafana-ui/src/components/Table/TableNG/hooks.ts

```diff
@@ -534,16 +534,30 @@ export function useColumnResize(
   return dataGridResizeHandler;
 }
 
-export function useScrollbarWidth(ref: RefObject<DataGridHandle>, height: number, renderedRows: TableRow[]) {
+export function useScrollbarWidth(ref: RefObject<DataGridHandle>, height: number) {
   const [scrollbarWidth, setScrollbarWidth] = useState(0);
 
   useLayoutEffect(() => {
     const el = ref.current?.element;
 
-    if (el) {
-      setScrollbarWidth(el.offsetWidth - el.clientWidth);
+    if (!el) {
+      return;
     }
-  }, [ref, height, renderedRows]);
+
+    const updateScrollbarDimensions = () => {
+      setScrollbarWidth(el.offsetWidth - el.clientWidth);
+    };
+
+    updateScrollbarDimensions();
+
+    // observe changes to the DOM structure, and re-calc scrollbar width on them
+    const mutationObserver = new MutationObserver(updateScrollbarDimensions);
+    mutationObserver.observe(el, { childList: true, subtree: true });
```

Comment:
we gotta be careful here with subtree and childList. virtualization mutates an insane amount of nodes, so this becomes very expensive. i dont think we want to hook this up to the whole grid, we might be okay just attaching this to the header row which is mostly static except during occasional width or filter/sort changes.

Reply:
That's a good idea - I'll also test whether zero-height headers work (which is how "hide header" is implemented, we still write it into the DOM which is convenient in this situation)

Reply:
> which is how "hide header" is implemented

i would be okay if this did not work when headers are disabled if the implementation needed to be 2x more complex to support that very rare case.

Reply:
I played with it a bit with the header. It mostly worked with the header, including with header hidden, but then I realized that both the header approach as well as the full table approach failed in some scenarios, and that a resize observer honestly just makes more sense and works all the time.